### PR TITLE
kv: fix handling of column default values

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve? 

Properly fill in the default value for a missing column.

### What is changed and how it works?

* if the column is AUTO_INCREMENT, fill in with row_id (assume it is missing for the entire table instead of just a few values)
* if the column has DEFAULT, fill in that value
* (if the column is NULL, fill in NULL.)

### Check List 

Tests 

 - Integration test

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by GregoryIan** [comment]
LGTM

File: lightning/kv/sql2kv.go [hunk 1]
```diff
@@ -69,13 +69,16 @@ func (kvcodec *TableKVEncoder) Encode(
...
 			value, err = table.CastValue(kvcodec.se, row[j], col.ToInfo())
 			if err == nil {
 				value, err = col.HandleBadNull(value, kvcodec.se.vars.StmtCtx)
 			}
+		} else if mysql.HasAutoIncrementFlag(col.Flag) {
+			// we still need a conversion, e.g. to catch overflow with a TINYINT column.
+			value, err = table.CastValue(kvcodec.se, types.NewIntDatum(rowID), col.ToInfo())
 		} else {
-			value, err = table.GetColOriginDefaultValue(kvcodec.se, col.ToInfo())
+			value, err = table.GetColDefaultValue(kvcodec.se, col.ToInfo())
```
Comment (csuzhangxc) [inline lightning/kv/sql2kv.go]: not handle some cases like `current_timestamp` anymore?

Comment (kennytm) [reply lightning/kv/sql2kv.go]: `CURRENT_TIMESTAMP` does work, it currently fetches the actual current time 🤔. (Before this PR it was always NULL.) Gonna change to a fixed timestamp stored in the checkpoint to ensure determinism in a future PR.

**Pull request merged**